File: Java/wellmind-app/src/main/java/com/wellmind/dto/registrobemestar/RegistroBemestarDTO.java
package com.wellmind.dto.registrobemestar;

import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RegistroBemestarDTO {

    private Long idRegistro;
    private Long idUsuario;
    private String nomeUsuario;
    private Integer nivelHumor;
    private Integer nivelEstresse;
    private Integer nivelEnergia;
    private Double horasSono;
    private Integer qualidadeSono;
    private Double mediaBemestar;
    private String classificacao;
    private String observacoes;
    private LocalDateTime dataRegistro;
    private Boolean temAlerta;
    private String motivoAlerta;
}

File: DotNet/WellMindApi/src/WellMindApi.Application/DTOs/Registros/RegistroBemEstarDto.cs
namespace WellMindApi.Application.DTOs.Registros;

/// <summary>
/// DTO de retorno de Registro de Bem-Estar
/// </summary>
public record RegistroBemEstarDto
{
    public int IdRegistro { get; init; }
    public int IdUsuario { get; init; }
    public DateTime DataRegistro { get; init; }
    public int NivelHumor { get; init; }
    public int NivelEstresse { get; init; }
    public int NivelEnergia { get; init; }
    public decimal HorasSono { get; init; }
    public int QualidadeSono { get; init; }
    public string? Observacoes { get; init; }

    // Campos calculados
    public decimal IndiceBemEstar { get; init; }
    public string StatusSaude { get; init; } = string.Empty;
    public bool IndicaBurnout { get; init; }
    public string AreaCritica { get; init; } = string.Empty;
    public string Recomendacao { get; init; } = string.Empty;
}
